libs/pageserver_api/src/key.rs

```diff
@@ -144,3 +144,35 @@ impl Key {
 pub fn is_rel_block_key(key: &Key) -> bool {
     key.field1 == 0x00 && key.field4 != 0
 }
+
+impl std::str::FromStr for Key {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        Self::from_hex(s)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use crate::key::Key;
+
+    #[test]
+    fn display_fromstr_bijection() {
+        let mut rng = rand::thread_rng();
```

Comment:
nit (because this test is unlikely to suffer any issues), but: please use a fixed seed for unit tests -- in cases where we just want a "foo" number, it's nice to have the same one each time to keep tests reproducible (imagine if you had a serialization but that only affected zeros, or something like that)

Reply:
moved to https://github.com/neondatabase/neon/pull/6145/files#r1427990161